tests/FSharp.Compiler.ComponentTests/ErrorMessages/TypeMismatchTests.fs

```diff
@@ -7,6 +7,52 @@ open FSharp.Test.Compiler
 
 module ``Type Mismatch`` =
 
+    module ``Different tuple lengths`` =
+        
+        [<Fact>]
+        let ``Known type on the left``() =
+            FSharp """
+let x: int * int * int = 1, ""
+let x: int * string * int = "", 1
+let x: int * int = "", "", 1
+            """
+            |> typecheck
+            |> shouldFail
+            |> withDiagnostics [
+                (Error 1, Line 2, Col 26, Line 2, Col 31,
+                 "Type mismatch. Expecting a\n    'tuple of length 3 (int * int * int)'    \nbut given a\n    'tuple of length 2'    \n")
```

Comment:
I suggest

    Type mismatch. Expecting a tuple of length 3 of type
        int * int * int
    but given a tuple of length 2 

The diagnostic should read fluently as English without the English fragments on new lines nor in quotes.  It's often helpful to fully spec out a full matrix of generated diagnostics separately before coding them.

That said I'm in two minds about whether this represents an improvement over 

    Type mismatch. Expecting a tuple of length 3 of type
        int * int * int
    but given a tuple of length 2 of type
        'a * 'b

In the case above showing the r.h.s. type is definitely preferable but I'm loathe to do anything that doesn't involve immediate emit of the diagnostic.

Just to not we're very inconsistent about whether types get put in quotes but in this case we were doing it before so we should do it here.  We should really colorize the outputs or something, or use unicode.

Reply:
Personally, I find it confusing when we show no concrete types in case when they're known.

But we definitely should start enriching out messages with some pointers and add colouring.

Reply:
Yeah I think you're right.

Reply:
> But we definitely should start enriching out messages with some pointers and add colouring.

Could be also very cool to make types themselves available to the tools 🙂